app/routes.py

```diff
@@ -1 +1,229 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, make_response, request, abort, jsonify 
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+#to call slack api in this file:
+from dotenv import load_dotenv
+import requests 
+import os
+load_dotenv()
+
+# Blueprints:
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} invalid"}, 400))
+    
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message": f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+
+@tasks_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+    if not "title" in request_body or not "description" in request_body:
+        # return make_response({"details": "Invalid data"}, 400)
+        abort(make_response({"details": "Invalid data"}, 400))
+    if not "completed_at" in request_body:
+        request_body["completed_at"] = None
+
+    new_task = Task.from_dict(request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    tasks_response = []
+    # Getting the query params: If there's no "sort" get all tasks
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        total_tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_query == "desc":
+        total_tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        total_tasks = Task.query.all()
+    
+    for task in total_tasks:
+        tasks_response.append(task.to_dict())
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=['GET'])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.to_dict()}, 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return {"task": task.to_dict()}, 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return {
+        "details": f'Task {task_id} \"{task.title}\" successfully deleted'
```

Comment:
```suggestion
        "details": f'Task {task_id} "{task.title}" successfully deleted'
```
Since you switched to single quotes for the f-string, you don't need to escape out the double quotation marks.

If I'm being really picky, I'd probably recommend switching to having double quotes for the f-string and single quotes around task title to be consistent with your other strings.